src/components/Shared/Table.js

```diff
@@ -0,0 +1,186 @@
+import React, { useState, useMemo } from "react";
+import {
+  useTable,
+  useGlobalFilter,
+  useSortBy,
+} from "react-table";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import Input from "./Input";
+
+const Table = styled.div`
+  width: 100%;
+  display: table;
+`;
+
+const TableHeader = styled.div`
+  width: 100%;
+  background: ${({ theme }) => theme.color.grey};
+  font-family: "Bebas Neue";
+  font-size: ${({ theme }) => theme.fontSize.middle};
+  color: ${({ theme }) => theme.color.white};
+
+  .tr {
+    width: 100%;
+    display: table;
+    table-layout: fixed;
+  }
+
+  .th {
+    padding: ${({ theme }) => theme.padding.small};
+    display: table-cell;
+    text-align: center;
+
+    &.sort-asc::after {
+      padding: 0 0 0 0.5rem;
+      font-size: 0.4rem;
+      content: "▼";
+      vertical-align: middle;
+    }
+
+    &.sort-desc::after {
+      padding: 0 0 0 0.5rem;
+      font-size: 0.4rem;
+      content: "▲";
+      vertical-align: middle;
+    }
+  }
+`;
+
+const TableBody = styled.div`
+  width: 100%;
+  background: ${({ theme }) => theme.color.darkgrey};
+  font-size: ${({ theme }) => theme.fontSize.base};
+  color: ${({ theme }) => theme.color.white};
+  height: ${(props) => props.height};
+  overflow-y: auto;
+
+  .tr {
+    width: 100%;
+    border-bottom: 1px solid ${({ theme }) => theme.color.black};
+    display: table;
+    table-layout: fixed;
+  }
+
+  .td {
+    padding: ${({ theme }) => theme.padding.small};
+    border-right: 1px solid ${({ theme }) => theme.color.black};
+    display: table-cell;
+    text-align: center;
+
+    &:last-child {
+      border-right: none;
+    }
+  }
+`;
+
+function SharedTable({
+  tableColumns,
+  tableData,
+  search,
+  colWidths,
+  tableHeight,
+  placeholder = "",
+}) {
```

Comment:
구조분해할당으로 받을 객체 속성들이 많으면, 파라미터는 props로 설정한 후 함수 안에서 구조분해할당 해주세요.
```suggestion
function SharedTable(props) {
  {
    tableColumns,
    tableData,
    ...,
  } = props;
```